// @sanity/runtime-cli/dist/actions/blueprints/operations.js
import config from '../../config.js';
const { blueprints } = config.server;
export const stacksUrl = `${blueprints}vX/blueprints/stacks`;
function getHeaders(projectId) {
    return {
        Authorization: `Bearer ${config.token}`,
        'Content-Type': 'application/json',
        'X-Sanity-Scope-Type': 'project',
        'X-Sanity-Scope-Id': projectId,
    };
}
export async function getOperation({ stackId, operationId, projectId, }) {
    const path = `${stacksUrl}/${stackId}/operations/${operationId}`;
    const response = await fetch(path, {
        method: 'GET',
        headers: getHeaders(projectId),
    });
    if (!response.ok) {
        const errorText = await response.text();
        return {
            ok: false,
            error: errorText || 'Failed to fetch operation details',
            operation: null,
        };
    }
    const operation = await response.json();
    return {
        ok: true,
        error: null,
        operation,
    };
}
export async function listOperations({ stackId, projectId, }) {
    const path = `${stacksUrl}/${stackId}/operations`;
    const response = await fetch(path, {
        method: 'GET',
        headers: getHeaders(projectId),
    });
    if (!response.ok) {
        const errorText = await response.text();
        return {
            ok: false,
            error: errorText || 'Failed to fetch operations',
            operations: [],
        };
    }
    const operations = await response.json();
    return {
        ok: true,
        error: null,
        operations,
    };
}
